core/scheduler/service/replay_service.go

```diff
@@ -84,6 +102,25 @@ func (r *ReplayService) CreateReplay(ctx context.Context, tenant tenant.Tenant,
 	return replayID, nil
 }
 
+func (r *ReplayService) injectJobConfigWithNamespaceConfig(ctx context.Context, tenant tenant.Tenant, config *scheduler.ReplayConfig) error {
+	// get namespace configuration to obtain the execution project specifically for replay.
+	namespaceDetails, err := r.namespaceRepo.GetByName(ctx, tenant.ProjectName(), tenant.NamespaceName())
+	if err != nil {
+		return err
+	}
+
+	for cfgKey, cfgVal := range namespaceDetails.GetConfigs() {
+		// only inject namespace-level configs for replay if they are not present in the ReplayConfig
+		if overridedConfigKey, found := namespaceConfigForReplayMap[cfgKey]; found {
+			if _, configExists := config.JobConfig[overridedConfigKey]; !configExists {
+				config.JobConfig[overridedConfigKey] = cfgVal
```

Comment:
let's not to mutate the config object, unless we have strong reason to do it
https://luzkan.github.io/smells/mutable-data

let's return as a new object instead

Reply:
agree, and since it only modifies JobConfig, I will return the map[string]string